src/System.Drawing.Common/tests/IconTests.cs

```diff
@@ -0,0 +1,614 @@
+            {
+                stream.Write(bytes, 0, bytes.Length);
+
+                stream.Position = 0;
+                Assert.Throws(exceptionType, () => new Icon(stream));
+            }
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [MemberData(nameof(Size_TestData))]
+        public void Ctor_Icon_Width_Height(string fileName, Size size, Size expectedSize)
+        {
+            var sourceIcon = new Icon(Helpers.GetEmbeddedResourcePath(fileName));
+            var icon = new Icon(sourceIcon, size.Width, size.Height);
+            Assert.Equal(expectedSize.Width, icon.Width);
+            Assert.Equal(expectedSize.Height, icon.Height);
+            Assert.Equal(expectedSize, icon.Size);
+            Assert.NotSame(sourceIcon.Handle, icon.Handle);
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [MemberData(nameof(Size_TestData))]
+        public void Ctor_Icon_Size(string fileName, Size size, Size expectedSize)
+        {
+            var sourceIcon = new Icon(Helpers.GetEmbeddedResourcePath(fileName));
+            var icon = new Icon(sourceIcon, size);
+            Assert.Equal(expectedSize.Width, icon.Width);
+            Assert.Equal(expectedSize.Height, icon.Height);
+            Assert.Equal(expectedSize, icon.Size);
+            Assert.NotSame(sourceIcon.Handle, icon.Handle);
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void Ctor_NullIcon_ThrowsArgumentNullException()
+        {
+            Assert.Throws<ArgumentException>(null, () => new Icon((Icon)null, 32, 32));
+            Assert.Throws<ArgumentException>(null, () => new Icon((Icon)null, new Size(32, 32)));
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void Ctor_Type_Resource()
+        {
+            var icon = new Icon(typeof(IconTests), "48x48_multiple_entries_4bit.ico");
+            Assert.Equal(32, icon.Height);
+            Assert.Equal(32, icon.Width);
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void Ctor_NullType_ThrowsNullReferenceException()
+        {
+            Assert.Throws<NullReferenceException>(() => new Icon(null, "48x48_multiple_entries_4bit.ico"));
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [InlineData(typeof(Icon), null)]
+        [InlineData(typeof(Icon), "")]
+        [InlineData(typeof(Icon), "48x48_multiple_entries_4bit.ico")]
+        [InlineData(typeof(IconTests), "48x48_MULTIPLE_entries_4bit.ico")]
+        public void Ctor_InvalidResource_ThrowsArgumentNullException(Type type, string resource)
```

Comment:
The name of the method should be `Ctor_InvalidResource_ThrowsArgumentException` (not ArgumentNullException).